File: Ch03/03_05/head.go
package main

import (
	"fmt"
	"os"
)

func fileHead(fileName string, size int) ([]byte, error) {
	file, err := os.Open(fileName)
	if err != nil {
		return nil, err
	}
	defer file.Close()

	buf := make([]byte, size)
	n, err := file.Read(buf)

	if err != nil {
		return nil, err
	}
	if n != size {
		return nil, fmt.Errorf("%q too small", fileName)
	}

	return buf, nil
}

func main() {
	data, err := fileHead("head.png", 8)
	if err != nil {
		fmt.Println("ERROR:", err)
	} else {
		fmt.Println(data)
	}
}

File: Ch03/03_05/head.py
def file_head(file_name, size):
    with open(file_name, 'rb') as fp:
        data = fp.read(size)

    if len(data) < size:
        raise ValueError(f'{file_name!r} too small')

    return data


if __name__ == '__main__':
    data = file_head('Ch03/03_05/head.png', 8)
    print(data)
